src/Datadog.Trace.Ci.Shared/GitInfo.cs

```diff
@@ -439,44 +439,62 @@ public static bool TryGetFromObjectFile(string filePath, out GitCommitObject com
-                        }
+                            // Move to the offset of the object
+                            fs.Seek(packageOffset.Offset, SeekOrigin.Begin);
+                            byte[] packData = br.ReadBytes(2);
 
-                        using (var ms = new MemoryStream(packData, 2, packData.Length - 2))
-                        using (var defStream = new DeflateStream(ms, CompressionMode.Decompress))
-                        {
-                            byte[] buffer = new byte[8192];
-                            int readBytes = defStream.Read(buffer, 0, buffer.Length);
-                            defStream.Close();
-                            string strContent = Encoding.UTF8.GetString(buffer, 0, readBytes);
-                            commitObject = new GitCommitObject(strContent);
-                            return true;
+                            // Extract the object size
+                            int objectSize = (int)(packData[0] & 0x0F);
+
+                            if (packData[0] > 128)
+                            {
+                                objectSize += (packData[1] & 0x7F) * 16;
+                                if (packData[1] > 128)
+                                {
+                                    int multiplier = 128;
+                                    byte pData = br.ReadByte();
+                                    objectSize += (pData & 0x7F) * multiplier;
+                                    while (pData > 128)
+                                    {
+                                        multiplier *= 128;
+                                        pData = br.ReadByte();
+                                        objectSize += (pData & 0x7F) * multiplier;
+                                    }
+                                }
+                            }
+
+                            // Advance 2 bytes to skip the zlib magic number
+                            uint zlibMagicNumber = br.ReadUInt16();
+                            if ((byte)zlibMagicNumber == 0x78)
+                            {
+                                // Read the git commit object
+                                using (var defStream = new DeflateStream(br.BaseStream, CompressionMode.Decompress))
+                                {
+                                    byte[] buffer = new byte[objectSize];
+                                    int readBytes = defStream.Read(buffer, 0, buffer.Length);
+                                    defStream.Close();
+                                    string strContent = Encoding.UTF8.GetString(buffer, 0, readBytes);
+                                    commitObject = new GitCommitObject(strContent);
+                                    return true;
+                                }
+                            }
                         }
                     }
                 }
+                catch
```

Comment:
Are we ok throwing away the exception here? No logging or anything?

Reply:
Yeah, I didn't want to add a dependency to Datadog.Trace assembly for the logger, but I think I will.
The method is a `Try...` so I should keep the try/catch.

Reply:
done in afe9ff7